dev/reference/transactions/transaction-pessimistic.md

```diff
@@ -0,0 +1,100 @@
+
+```
+[pessimistic-txn]
+enable = true
+```
+
+When `enable` is set to `true`, the default transaction mode in TiDB is still optimistic. To enable the pessimistic transaction mode, choose any of the following methods that suits your needs:
+
+- Use `BEGIN PESSIMISTIC;` statement to start the transaction in the pessimistic transaction mode. Write in comment style as `BEGIN /*!90000 PESSIMISTIC */;` to make it compatible with the MySQL syntax.
+
+- Execute the `set @@tidb_txn_mode = 'pessimistic';` statement to allow all the transactions processed in this session to be in the pessimistic transaction mode.
+
+- Enable the pessimistic transaction mode in the configuration file. This allows all transactions (except auto-committed single-statement ones) to adopt the pessimistic transaction mode.
+
+    ```
+    [pessimistic-txn]
+    enable = true
+    default = true
+    ```
+
+If the pessimistic transaction mode is enabled in the configuration file by default, use one of the following methods to adopt the optimistic transaction mode for the transaction:
+
+- Use `BEGIN OPTIMISTIC;` statement to start the transaction in the optimistic transaction mode. Write in comment style as `BEGIN /*!90000 OPTIMISTIC */;` to make it compatible with the MySQL syntax.
+
+- Execute `set @@tidb_txn_mode = 'optimistic';` statement to allow all the transactions processed in this session to be in the optimistic transaction mode.
+
+## Enablement Priority
+
+- `BEGIN PESSIMISTIC;` and `BEGIN OPTIMISTIC;` are statements with the highest priority.
+
+-  Session variable `tidb_txn_mode` has the second highest priority.
+
+- You can configure the `default` file. If the transaction starts with `BEGIN` statement and the value of `tidb_txn_mode` is an empty string, the transaction mode is determined by the `default` in the configuration file.
+
+## Configuration Parameter
+
+The related configuration file is under the `[pessimistic-txn]` file. Besides the `enable` and `default` arguments, you can also configure the following arguments:
+
+- `ttl`
+
+    ```
+    ttl = "30s"
+    ```
+
+    The default value of the pessimistic locking timeout `ttl` is 30s. The value configured must be between "15s" and "60s". Otherwise, the error will be reported.
+
+    The transaction will fail if its execution time exceeds the value of `ttl`. If the timeout value is set too high, it might cause long waiting time of write transaction when `tidb-server` fails. If too low, the transaction might be rolled back due to the conflict with other transaction.
+
+- `max-retry-count`
+
+    ```
+    max-retry-count = 256
+    ```
+
+    In the pessimistic transaction mode, the transaction can automatically retry a single statement. You can configure the maximum number of automatically retry to avoid the extreme case that the single statement is run endlessly. Normally you do not need to modify this configuration.
```

Comment:
```suggestion
    A pessimistic transaction can automatically retry a single statement. You can specify the maximum retrying times by setting the parameter `max-retry-count` to avoid retrying a statement endlessly in some extreme cases. Normally, you do not need to modify this configuration.
```